File: 724-find-pivot-index/724-find-pivot-index.rs
impl Solution {
    pub fn pivot_index(mut nums: Vec<i32>) -> i32 {
        let sum = nums.iter().sum::<i32>();
        let mut leftsum = 0;
        for i in 0..nums.len() {
            if (2*leftsum + nums[i] == sum) {
                return i as i32;
            }
            leftsum += nums[i];
        }
        return -1
    }
}

File: 724-find-pivot-index/724-find-pivot-index.ts
function pivotIndex(nums: number[]): number {
    let sum = nums.reduce((acc, a) => acc + a, 0);
    let acc = 0;
    for (let i = 0; i < nums.length; i++) {
        if (sum - nums[i] == 2*acc) {
            return i;
        }
        acc += nums[i];
    }
    return -1;
};